test/src/tests-forwarders.js

```diff
@@ -2352,42 +2642,70 @@ describe('forwarders', function() {
 
         done();
     });
+    });
+    });
+    });
 
     it('should save logged in status of most recent user to cookies when logged in', function(done) {
-        mParticle._resetForTests(MPConfig);
-
-        mockServer.respondWith(urls.identify, [
-            200,
-            {},
-            JSON.stringify({ mpid: 'MPID1', is_logged_in: true }),
-        ]);
+        fetchMockSuccess(urls.identify, {
+            mpid: 'MPID1',
+            is_logged_in: true,
+        });
 
         mParticle.init(apiKey, window.mParticle.config);
+        waitForCondition(() => {
+            return (
+                window.mParticle.getInstance()._Store.identityCallInFlight === false
+            );
+        })
+        .then(() => {
         const ls = mParticle.getInstance()._Persistence.getLocalStorage();
         ls.l.should.equal(true);
-
+        
         mParticle.init(apiKey, window.mParticle.config);
+        waitForCondition(() => {
+            return (
+                window.mParticle.getInstance()._Store.identityCallInFlight === false
+            );
+        })
+        .then(() => {
         const ls2 = mParticle.getInstance()._Persistence.getLocalStorage();
         ls2.hasOwnProperty('l').should.equal(true);
-
-        mockServer.respondWith(urls.logout, [
-            200,
-            {},
-            JSON.stringify({ mpid: 'MPID1', is_logged_in: false }),
-        ]);
-
+        
+        fetchMockSuccess(urls.logout, {
+            mpid: 'MPID1', is_logged_in: false
+        });
+        
         mParticle.Identity.logout();
+        waitForCondition(() => {
+            return (
+                window.mParticle.getInstance()._Store.identityCallInFlight === false
+            );
+        })
+        .then(() => {
         const ls3 = mParticle.getInstance()._Persistence.getLocalStorage();
         ls3.l.should.equal(false);
 
         mParticle.init(apiKey, window.mParticle.config);
+        waitForCondition(() => {
+            return (
+                window.mParticle.getInstance()._Store.identityCallInFlight === false
+            );
+        })
+        .then(() => {
+
         const ls4 = mParticle.getInstance()._Persistence.getLocalStorage();
         ls4.l.should.equal(false);
 
         done();
     });
+    });
+    });
+    });
+    });
 
     it('should not set integration attributes on forwarders when a non-object attr is passed', function(done) {
+        mParticle.init(apiKey, window.mParticle.config)
```

Comment:
Why is `init` begin added here?

Reply:
it was removed from the beforeeach, and is only needed in some places, such as this test